tesseract/tesseract_urdf/test/tesseract_urdf_unit.cpp

```diff
@@ -0,0 +1,2231 @@
+  }
+
+  {
+    std::string str = "<link name=\"my_link\">"
+                      "  <visual>"
+                      "    <origin xyz=\"0 0 0\" rpy=\"0 0 0\" />"
+                      "    <geometry>"
+                      "      <box size=\"1 1 1\" />"
+                      "    </geometry>"
+                      "    <material name=\"Cyan\">"
+                      "      <color rgba=\"0 1.0 1.0 1.0\"/>"
+                      "    </material>"
+                      "  </visual>"
+                      "</link>";
+    tesseract_scene_graph::Link::Ptr elem;
+    auto status =
+        runTest<tesseract_scene_graph::Link::Ptr>(elem, str, "link", locateResource, empty_available_materials);
+    EXPECT_TRUE(*status);
+    EXPECT_TRUE(elem->getName() == "my_link");
+    EXPECT_TRUE(elem->inertial == nullptr);
+    EXPECT_TRUE(elem->visual.size() == 1);
+    EXPECT_TRUE(elem->collision.size() == 0);
+  }
+
+  {
+    std::string str = "<link name=\"my_link\">"
+                      "  <collision>"
+                      "    <origin xyz=\"0 0 0\" rpy=\"0 0 0\"/>"
+                      "    <geometry>"
+                      "      <cylinder radius=\"1\" length=\"0.5\"/>"
+                      "    </geometry>"
+                      "  </collision>"
+                      "</link>";
+    tesseract_scene_graph::Link::Ptr elem;
+    auto status =
+        runTest<tesseract_scene_graph::Link::Ptr>(elem, str, "link", locateResource, empty_available_materials);
+    EXPECT_TRUE(*status);
+    EXPECT_TRUE(elem->getName() == "my_link");
+    EXPECT_TRUE(elem->inertial == nullptr);
+    EXPECT_TRUE(elem->visual.size() == 0);
+    EXPECT_TRUE(elem->collision.size() == 1);
+  }
+
+  {
+    std::string str = "<link name=\"my_link\"/>";
+    tesseract_scene_graph::Link::Ptr elem;
+    auto status =
+        runTest<tesseract_scene_graph::Link::Ptr>(elem, str, "link", locateResource, empty_available_materials);
+    EXPECT_TRUE(*status);
+    EXPECT_TRUE(elem->getName() == "my_link");
+    EXPECT_TRUE(elem->inertial == nullptr);
+    EXPECT_TRUE(elem->visual.size() == 0);
+    EXPECT_TRUE(elem->collision.size() == 0);
+  }
+
+  {
+    std::string str = "<link >"
+                      "  <visual>"
+                      "    <origin xyz=\"0 0 0\" rpy=\"0 0 0\" />"
+                      "    <geometry>"
+                      "      <box size=\"1 1 1\" />"
+                      "    </geometry>"
+                      "    <material name=\"Cyan\">"
+                      "      <color rgba=\"0 1.0 1.0 1.0\"/>"
+                      "    </material>"
+                      "  </visual>"
+                      "</link>";
+    tesseract_scene_graph::Link::Ptr elem;
+    auto status =
+        runTest<tesseract_scene_graph::Link::Ptr>(elem, str, "link", locateResource, empty_available_materials);
+    EXPECT_FALSE(*status);
+  }
+
+  {
+    std::string str = "<link name=\"my_link\">"
+                      "  <visual>"
+            
```

Comment:
I am not following?

Reply:
This is a case that is supposed to fail. Should this be logged as debug or info instead since it is not actually an error? I also realize that we often log important things as errors just so that they grab our attention, so that is fine as well.